### PR TITLE
Convert patternfly.form module directives to components.

### PR DESCRIPTION
**Pull request by jeff-phillips-18**

### CODE REVIEW

**Comment by jeff-phillips-18** [comment]
@dgutride @dtaylor113 @dlabrecq

**Comment by dtaylor113** [comment]
LGTM.  Whjy are things moved into 'src/form/examples/', directives in 'examples' seem to be core directives (form-buttons, form-group, remain char count.), not examples.

**Comment by jeff-phillips-18** [comment]
Only the ng-doc examples where moved there. I think this makes it easier to read the real component code and not have to look past all the ng-doc stuff.

**Comment by waldenraines** [comment]
I know that you didn't write this originally but what are your thoughts on changing the form components to use normal BS3 form markup instead of `.form-horizontal` markup?

**Comment by jeff-phillips-18** [comment]
Sorry, what is abnormal about form-horizontal?

**Comment by waldenraines** [comment]
> Sorry, what is abnormal about form-horizontal?

It's not that there is anything abnormal about it but I think the default should be the normal form and if we want a horizontal form we should have a different directive or different option for that.

**Comment by jeff-phillips-18** [comment]
Probably a better way to go. Currently we default classes for the label and input components which really force the form-horizontal use. If we don't do that, the application could use either but will have to set the pfLabelClass and pfInputClass to use horizontal.

**Comment by waldenraines** [comment]
> Probably a better way to go. Currently we default classes for the label and input components which really force the form-horizontal use. If we don't do that, the application could use either but will have to set the pfLabelClass and pfInputClass to use horizontal.

Exactly.  This was originally copy/pasted from Katello where we were using horizontal and never updated.

File: src/form/examples/form-group.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/**
+ * @ngdoc directive
+ * @name patternfly.form.directive:pfFormGroup
+ *
```
Comment (dgutride) [inline src/form/examples/form-group.js]: restrict E should be in the header

File: src/form/form-buttons/form-buttons.component.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    form: '^form'
+  },
+  templateUrl: 'form/form-buttons/form-buttons.html',
+  controller: function () {
+    'use strict';
+
+    var ctrl = this;
+
+    if (ctrl.pfWorking === undefined) {
+      ctrl.pfWorking = false;
```
Comment (dgutride) [inline src/form/form-buttons/form-buttons.component.js]: Generally any initialization code should be moved to $onInit

File: src/form/form-group/form-group.component.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    pfInputClass: '@'
+
+  },
+  require: {
+    form: '^form'
+  },
+  transclude: true,
+  templateUrl: 'form/form-group/form-group.html',
+
+  controller: function ($scope, $element) {
```
Comment (dgutride) [inline src/form/form-group/form-group.component.js]: $scope should be removed from the function

**Comment by jeff-phillips-18** [comment]
@dgutride Remaining characters count is not a component. Not sure how we would convert it.

**Comment by jeff-phillips-18** [comment]
@dgutride  Updated per your comments.

File: src/form/examples/form-group.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+           <input id="name" name="name" ng-model="item.name" type="text" required/>
+         </pf-form-group>
+         <pf-form-group pf-label="Description">
+           <textarea id="description" name="description" ng-model="item.description">
+             {{ item.description }}
+           </textarea>
+         </pf-form-group>
+       </form>
+       <p>Horizontal Form</p>
+       <form class="form-horizontal">
```
Comment (waldenraines) [inline src/form/examples/form-group.js]: Should this be updated to reflect the discussed changes to `pf-form-group` (i.e. to remove the horizontal form specific classes)?

File: src/form/form-group/form-group.html [hunk 5]
```diff
@@ -1,9 +1,9 @@
-<div class="form-group" ng-class="{ 'has-error' : hasErrors() }">
-  <label for="{{ pfField }}" class="control-label {{ pfLabelClass }}">{{ pfLabel }}</label>
-  <div class="{{ pfInputClass }}">
+<div class="form-group" ng-class="{ 'has-error' : $ctrl.hasErrors() }">
+  <label for="{{ $ctrl.pfField }}" class="control-label {{ $ctrl.pfLabelClass }}">{{ $ctrl.pfLabel }}</label>
```
Comment (waldenraines) [inline src/form/form-group/form-group.html]: Should this be updated to reflect the discussed changes to `pf-form-group` (i.e. to remove the horizontal form specific classes)?

**Comment by waldenraines** [comment]
Ah, missed the boat.  I thought we were going to change the form groups to not be `.form-horizontal` centric?

**Comment by jeff-phillips-18** [comment]
The are no longer .form-horizontal centric.

File: src/form/form-group/form-group.html (hunk 5 above)
Comment (jeff-phillips-18) [inline src/form/form-group/form-group.html]: This is the example code. It shows both normal and horizontal forms:

image

**Comment by waldenraines** [comment]
> The are no longer .form-horizontal centric.

👍

**Pull request merged**